### PR TITLE
Agent size experiment

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR swaps agents over to use an experimental pool I've created which has a difference VM size to see if it makes a difference in execution times for tests.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -147,11 +147,11 @@ jobs:
       maxParallel: $[ variables['MaxParallelTestJobs'] ]
       matrix:
         Linux:
-          Pool: azsdk-pool-mms-ubuntu-1804-general
+          Pool: azsdk-pool-mms-ubuntu-1804-experimental
           OSVmImage: MMSUbunut18.04
           TestTargetFramework: netcoreapp2.1
         Windows_NetCoreApp:
-          Pool: azsdk-pool-mms-win-2019-general # Comment out to swap back to public hosted pool.
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:154]: nit: might want to remove or change these comments now.

Comment (mitchdenny) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:154]: Yeah, I'll do some further clean-up here. This PR will never get merged though, its for testing purpsoes.

**Pull request closed without merging**